irepela/homework_8/warrior.py

```diff
@@ -0,0 +1,39 @@
+# Details:  https://py.checkio.org/mission/the-warriors/share/e171c1a99e4e63b4551cfd820252edcd/
+
+
+class Warrior:
+
+    def __init__(self):
+        self.health = 50
+        self.attack = 5
+        self.is_alive = True
+
+    def hit(self, damage):
+        if self.health > 0:
+            self.health = self.health - damage
+        else:
+            self.is_alive = False
```

Comment:
if you have many places in code where hp is updated, you will have to modify there is_alive also
so you have dependancy on other param, use dynamic attr